### PR TITLE
feat: Commerce Coordinator step in retirement pipeline

### PR DESCRIPTION
**Pull request by JadeyOlivier**
Adding caller function for new Commerce Coordinator retirement step.

### CODE REVIEW

**Review by grmartin — APPROVED** [review]
I dont know if im the best one to approve this, but the tests pass so. lets go for it.

File: tubular/edx_api.py [hunk 1]
```diff
@@ -517,3 +517,18 @@ def retire_learner(self, learner):
...
+    Commerce-Coordinator API client.
+    """
+    @_retry_lms_api()
+    def retire_learner(self, learner):
+        """
+        Performs the learner retirement step for Commerce-Coordinator.
+        Passes the learner's LMS User Id instead of username.
+        """
+        data = {'edx_lms_user_id': learner['user']['id']}
+        api_url = self.get_api_url('lms/user_retirement')
```
Comment (pwnage101) [inline tubular/edx_api.py:533]: Do you plan to add the `lms/user_retirement` endpoint to the commerce-coordinator service? It currently doesn't seem to exist in the django urls.py routes: https://github.com/edx/commerce-coordinator/blob/main/commerce_coordinator/apps/lms/urls.py

File: tubular/tests/test_edx_api.py [hunk 2]
```diff
@@ -558,3 +558,39 @@ def test_retire_learner(self, mock_method):
...
+            POST,
+            urljoin(self.commerce_coordinator_base_url, 'lms/user_retirement'),
+            match=[matchers.json_params_matcher(json_data)]
+        )
+        self.commerce_coordinator_api.retire_learner(
+            learner=get_fake_user_retirement()
+        )
+        mock_method.assert_called_once_with(
+            learner=get_fake_user_retirement()
+        )
```
Comment (pwnage101) [inline tubular/tests/test_edx_api.py:596]: I know you just copy+pasted this from other tests, but LOL this is quite a thin test that doesn't actually run the retire_learner function.  I'm not going to block on this, but if you feel motivated---I think we'd ideally mock the underlying self._requests call.

**Review by pwnage101 — APPROVED** [review]

File: tubular/edx_api.py (hunk 1 above)
Comment (JadeyOlivier) [reply tubular/edx_api.py:533]: Yup! Will be merged in along with this work: https://github.com/edx/commerce-coordinator/pull/247/files#diff-d76063092b846765b0a4c6b15b4a31f3c0bad316aa7e39242bf7d9fcd422a088R19

File: tubular/tests/test_edx_api.py (hunk 2 above)
Comment (JadeyOlivier) [reply tubular/tests/test_edx_api.py:596]: Okay cool, I will make a note to create the new test once we've gotten this work through. Thank you!

**Pull request closed without merging**